dotnet/src/Agents/OpenAI/OpenAIAssistantDefinition.cs

```diff
@@ -1,30 +1,19 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
-using System.Collections.Generic;
 using System.Text.Json.Serialization;
 
 namespace Microsoft.SemanticKernel.Agents.OpenAI;
 
 /// <summary>
 /// Defines an assistant.
 /// </summary>
-public sealed class OpenAIAssistantDefinition
+public sealed class OpenAIAssistantDefinition : OpenAIAssistantCapabilities
```

Comment:
Is there a technical reason of this hierarchy?

Reply:
I wanted to eliminate the possiblility of conflicting / confusing definition of `Name`, `Description`, and `Instructions` what a `PromptTemplateConfig` is provided so I split out the core capabilities, sans these fields...but we still use `OpenAIAssistantDefinition` to describe a complete assistant.

Using inheritance to avoid code duplication between these classes and also how they are consumed in `AssistantCreationOptionsFactory`